datafusion/common/src/join_type.rs

```diff
@@ -28,21 +28,26 @@ use crate::{DataFusionError, Result};
 /// Join type
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Hash)]
 pub enum JoinType {
-    /// Inner Join
+    /// Inner Join - Returns rows where there is a match in both tables.
```

Comment:
if we want to get precise, this should also say that all matches are returned (n-to-m)